### PR TITLE
update readme with steps to create new release image

### PR DESCRIPTION
**Pull request by roshkhatri**
Update readme with steps to create new release image

## When to build and publish new Docker Image?
After a new major, minor or patch version of Valkey is released on https://github.com/valkey-io/valkey.

## How to build and publish new version Docker Image?
*Pre-requisites: [Fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo) this repo, create a private dockerhup repo and setup the Github secrets to access the private dockerhup repo.*
1. Validate if the metadata for the new version is updated at https://github.com/valkey-io/valkey-hashes/blob/main/README - If it is not updated, please open an issue here https://github.com/valkey-io/valkey-hashes/issues
2. If a new major or minor version is released, please create a new dir in the `valkey-container` repo. For example [`7.2`](https://github.com/valkey-io/valkey-container/tree/mainline/7.2).
3. Run the `update.sh` script locally, which will update the `versions.json` and also populate the Dockerfiles for the new versions in the respective directories. Running the `update.sh` file, executes the `versions.sh` which updates the `versions.json` file with the required metadata from https://github.com/valkey-io/valkey-hashes/blob/main/README. Once `versions.json` is updated, `apply-templates.sh` is executed which updates Docker files for all the versions directories in the repo (For example, [`7.2`](https://github.com/valkey-io/valkey-container/tree/mainline/7.2)).
4. Validate if the version and the info is populated correctly from the https://github.com/valkey-io/valkey-hashes/blob/main/README file.
5. See if all the tests pass on your fork and the your private docker hub has been updated.
5. Update the `dockerhub-description.md` with the updated tags in your private docker hub and the Docker files links.
6. Publish a PR with these changes. For example: https://github.com/valkey-io/valkey-container/pull/8
7. Once the PR is merged, Sit back, relax and enjoy looking at your creation getting published to the official Docker hub page.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,12 @@ This Project is the Git repo of the [Valkey "Official Image"](https://hub.docker
 The Project is now maintained by [the Valkey Community](https://github.com/valkey-io/valkey/) 
 and it was forked from https://github.com/docker-library/redis.
```
Comment (parthpatel) [inline README.md:4]: Do we really want to keep this reference? The other project is still an active project and probably under the same license as Redis.

File: README.md [hunk 2]
```diff
@@ -3,3 +3,12 @@ This Project is the Git repo of the [Valkey "Official Image"](https://hub.docker
...
 
+## How to build and publish new version Docker Image?
```
Comment (parthpatel) [inline README.md:9]: Also, when to build and publish new version docker image?

File: README.md [hunk 3]
```diff
@@ -3,3 +3,12 @@ This Project is the Git repo of the [Valkey "Official Image"](https://hub.docker
...
+*Pre-requisites: Fork this repo, create a private dockerhup repo and setup the Github secrets to access the private dockerhup repo.*
```
Comment (parthpatel) [inline README.md]: link https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo

File: README.md [hunk 4]
```diff
@@ -3,3 +3,12 @@ This Project is the Git repo of the [Valkey "Official Image"](https://hub.docker
...
+1. Validate if the metadata for the new version is updated at https://github.com/valkey-io/valkey-hashes/blob/main/README
```
Comment (parthpatel) [inline README.md]: What if the hashes file is not updated, but I want to release new docker image? May be explain the background behind each step?

File: README.md [hunk 5]
```diff
@@ -3,3 +3,12 @@ This Project is the Git repo of the [Valkey "Official Image"](https://hub.docker
...
+2. If its a new major version create a new dir. For example `7.2`.
```
Comment (parthpatel) [inline README.md]: create a new dir where? Does it matter if I create it in /tmp? Is 7.2 a hyperlink?

File: README.md [hunk 6]
```diff
@@ -3,3 +3,12 @@ This Project is the Git repo of the [Valkey "Official Image"](https://hub.docker
...
+3. Run the `update.sh` script locally, which will update the `versions.json` and also populate the Dockerfiles for the new versions in the repective directories.
```
Comment (parthpatel) [inline README.md]: All of this sh files have 0 documentation or comments. May be elaborate on what is happening? Why is it needed?

File: README.md [hunk 7]
```diff
@@ -3,3 +3,12 @@ This Project is the Git repo of the [Valkey "Official Image"](https://hub.docker
...
+4. Validate if the version and the info is populated correctly.
```
Comment (parthpatel) [inline README.md]: how do I validate this?

File: README.md [hunk 8]
```diff
@@ -3,3 +3,12 @@ This Project is the Git repo of the [Valkey "Official Image"](https://hub.docker
...
+5. See if all the tests pass on your fork and the your private docker up has been updated. 
+5. Publish a PR with these changes. For example: https://github.com/valkey-io/valkey-container/pull/8
+6. Once the PR is merged, Sit back, relax and enjoy looking at your creation getting publish to the official Docker hub page.
```
Comment (parthpatel) [inline README.md]: You should explain a little more here. What goes into getting this published to official docker hub page? Are there any failures to look out for? 
Also there is a typo - "getting published".

File: temp.sh [hunk 9]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
Comment (parthpatel) [inline temp.sh]: What is this file for? :)

**Review by parthpatel — COMMENTED** [review]
Hey, the README.md looks fine if everything succeeds in first attempt. But otherwise I don't understand the details, so asked a bunch of questions on PR.

File: README.md (hunk 2 above)
Comment (roshkhatri) [reply README.md:9]: When we have a new version release from valkey

File: temp.sh (hunk 9 above)
Comment (roshkhatri) [reply temp.sh]: this was a temp file where I was testing some stuff

File: README.md (hunk 4 above)
Comment (roshkhatri) [reply README.md]: I am working on automating this process, so once a new release is pushed the hashes will be updated

File: README.md (hunk 5 above)
Comment (roshkhatri) [reply README.md]: In the valkey-container directory. I will link the example PR

File: README.md (hunk 1 above)
Comment (roshkhatri) [reply README.md:4]: It is maintained by dicker community and is under BSD, we can keep it.

**Review by parthpatel — APPROVED** [review]
LGTM

**Pull request merged**